apps/test/dataCallbacksTests.js

```diff
@@ -0,0 +1,102 @@
+var sinon = require('sinon');
+var testUtils = require('./util/testUtils');
+var assert = testUtils.assert;
+testUtils.setupLocales('applab');
+
+var $ = require('jquery');
+var React = require('react');
+window.$ = $;
+window.jQuery = window.$;
+window.React = React;
+
+// used in design mode
+window.Applab = {
+  appWidth: 320,
+  appHeight: 480
+};
+
+window.dashboard = window.dashboard || {};
+
+var Applab = require('@cdo/apps/applab/applab');
+var applabCommands = require('@cdo/apps/applab/commands');
+var errorHandler = require('@cdo/apps/errorHandler');
+
+describe('createRecord callbacks', function () {
+  var xhr;
+  var lastRequest;
+
+  // Intercept all XHR requests, storing the last one
+  before(function () {
+    xhr = sinon.useFakeXMLHttpRequest();
+    xhr.onCreate = function (req) {
+      lastRequest = req;
+    };
+  });
+
+  after(function () {
+    lastRequest = null;
+    xhr.restore();
+  });
+
+  it('calls onSuccess for 200', function () {
+    var hitSuccess = false;
+    var hitError = false;
+
+    var myRecord = { name: 'Alice' };
+    applabCommands.createRecord({
+      table: 'mytable',
+      record: myRecord,
+      onSuccess: function () {
+        hitSuccess = true;
+      },
+      onError: function () {
+        hitError = true;
+      }
+    });
+
+    lastRequest.respond(200, { "Content-Type": "application/json" },
+      JSON.stringify(myRecord));
+    assert(hitSuccess);
+    assert(!hitError);
+  });
+
+  it('calls onError for 403', function () {
+    var hitSuccess = false;
+    var hitError = false;
+
+    var myRecord = { name: 'Alice' };
+    applabCommands.createRecord({
+      table: 'mytable',
+      record: myRecord,
+      onSuccess: function () {
+        hitSuccess = true;
+      },
+      onError: function () {
+        hitError = true;
+      }
+    });
+
+    lastRequest.respond(403, { "Content-Type": "application/json" },
+      JSON.stringify(myRecord));
```

Comment:
NIT:  The response body would actually be error text like 'A table may have at most 1000 rows'  rather than the record (here and below).